audio_api/services/tts_service.py

```
@@ -249,33 +420,64 @@ async def _save_wav_file(
-            await asyncio.to_thread(
-                self._write_wav_file, file_path, pcm_data, channels, rate, sample_width
+            # Use async file operations
+            await self._write_wav_file_async(
+                file_path, pcm_data, channels, rate, sample_width
             )
 
             return file_path
         except Exception as e:
             logger.error(f"Error saving WAV file: {str(e)}")
-            raise
+            raise TTSError(f"Failed to save audio file: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)